docs/syntax_and_semantics/literals/string.md

```diff
@@ -183,20 +183,31 @@ Leading whitespace is removed from the heredoc contents according to the number
   STRING
 ```
 
-It is possible to directly call methods on heredoc string literals, or use them inside parentheses:
+A heredoc can be used like any other expression but only the first line is treated as such and essentially holds its value. The final line may not contain any Crystal code.
+It is possible to directly call methods on heredoc string literals, or use them as arguments.
```

Comment:
I think it's still not clear what are the rules. Just examples aren't enough (that's why the issue was opened). I would say something like "after the heredoc identifier, and in that same line, anything that follow continues the original expression that came before the heredoc. It's as if the end of the heredoc identifier is the end of the string. However, the string contents come in subsequent lines.

Reply:
Yeah, that puts it much better.